homeassistant/components/seventeentrack/config_flow.py

```diff
@@ -0,0 +1,133 @@
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class SeventeenTrackConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """17track config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> SchemaOptionsFlowHandler:
+        """Get options flow for this handler."""
+        return SchemaOptionsFlowHandler(config_entry, OPTIONS_FLOW)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+
+        errors = {}
+        if user_input:
+            client = await self._get_client()
+
+            try:
+                if not await client.profile.login(
+                    user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                ):
+                    errors = {"base": "invalid_credentials"}
+            except SeventeenTrackError as err:
+                _LOGGER.error("There was an error while logging in: %s", err)
+                errors = {"base": "cannot_connect"}
+
+            if not errors:
+                account_id = client.profile.account_id
+                await self.async_set_unique_id(account_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_USERNAME],
+                    data=user_input,
+                    options={
+                        CONF_SHOW_ARCHIVED: DEFAULT_SHOW_ARCHIVED,
+                        CONF_SHOW_DELIVERED: DEFAULT_SHOW_DELIVERED,
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=USER_SCHEMA,
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_data: dict[str, Any]) -> ConfigFlowResult:
+        """Import 17Track config from configuration.yaml."""
+
+        client = await self._get_client()
+
+        try:
+            login_result = await client.profile.login(
+                import_data[CONF_USERNAME], import_data[CONF_PASSWORD]
+            )
+        except SeventeenTrackError as err:
+            _LOGGER.error("There was an error while logging in: %s", err)
+            return self.async_abort(reason="cannot_connect")
+
+        if not login_result:
+            _LOGGER.error("Invalid username and password provided")
+            return self.async_abort(reason="invalid_credentials")
+
+        account_id = client.profile.account_id
+
+        await self.async_set_unique_id(account_id)
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(
+            title=import_data[CONF_USERNAME],
+            data=import_data,
+            options={
+                CONF_SHOW_ARCHIVED: DEFAULT_SHOW_ARCHIVED,
+                CONF_SHOW_DELIVERED: DEFAULT_SHOW_DELIVERED,
+            },
+        )
```

Comment:
For the options in the import: dont forget people also have these fields set, so we should use that field from the YAML, otherwise set to default

Reply:
Done